File: core/src/main/java/de/qabel/core/repository/Transaction.java
package de.qabel.core.repository;

import de.qabel.core.repository.exception.TransactionException;

public interface Transaction extends AutoCloseable {
    /**
     * Tries to commit the current transaction.
     * Make sure to rollback when this operation fails.
     */
    void commit() throws TransactionException;

    /**
     * Rolls back the current transaction.
     */
    void rollback() throws TransactionException;
}

File: core/src/main/java/de/qabel/core/repository/Transaction.kt
package de.qabel.core.repository

import de.qabel.core.repository.exception.TransactionException

interface Transaction : AutoCloseable {
    /**
     * Tries to commit the current transaction.
     * Make sure to rollback when this operation fails.
     */
    @Throws(TransactionException::class)
    fun commit()

    /**
     * Rolls back the current transaction.
     */
    @Throws(TransactionException::class)
    fun rollback()
}
